### PR TITLE
Consistently use txOptions

### LINKED ISSUE #92: Remove hardcoded transaction objects.
- Accept txOptions as a parameter.
- Dont use hardcoded values.
- Use Mosaic.Uitls.sendTransaction, it already estimated the gas if its not provided in txOptions.

### PR DESCRIPTION
**Pull request by hobofan**
FIXES #92

~Doesn't completely fix #92, since `Mosaic.Utils.sendTransaction` is not exposed from `mosaic.js` right now.~

### CODE REVIEW

**Comment by schemar** [comment]
> REFERENCES #92
> 
> Doesn't completely fix #92, since `Mosaic.Utils.sendTransaction` is not exposed from `mosaic.js` right now.

So should we expose `sendTransaction()`?
What was the conclusion here @sarvesh-ost @deepesh-kn 🤔

**Comment by sarvesh-ost** [comment]
> > REFERENCES #92
> > Doesn't completely fix #92, since `Mosaic.Utils.sendTransaction` is not exposed from `mosaic.js` right now.
> 
> So should we expose `sendTransaction()`?
> What was the conclusion here @sarvesh-ost @deepesh-kn 🤔

I think we decided to get `Utils.sendTransaction` for the mosaic. 🤔

File: lib/helpers/stake/gateway_composer/StakeHelper.js [hunk 1]
```diff
@@ -122,17 +130,25 @@ class StakeHelper {
...
 
-    const txOptions = {
+    const defaultOptions = {
       from: facilitator,
       to: oThis.gatewayComposer,
-      gas: '500000',
     };
 
-    const contract = new web3.eth.Contract(valueTokenAbi, valueTokenContractAddress, txOptions);
+    if (txOptions) {
```
Comment (sarvesh-ost) [inline lib/helpers/stake/gateway_composer/StakeHelper.js]: Should we add validations for from address in  txOptions ?

File: lib/helpers/stake/gateway_composer/StakeHelper.js [hunk 2]
```diff
@@ -379,7 +394,6 @@ class StakeHelper {
     const defaultOptions = {
       from: owner,
       to: oThis.gatewayComposer,
-      gas: '7500000',
```
Comment (sarvesh-ost) [inline lib/helpers/stake/gateway_composer/StakeHelper.js]: Why tx-options are required for raw tx? 🤔

**Comment by benjaminbollen** [comment]
This ticket needs a new assignee

**Comment by schemar** [comment]
@hobofan if it hasn't been done already, please expose `Utils.sendTransaction()`.

**Comment by schemar** [comment]
https://github.com/OpenSTFoundation/mosaic.js/issues/132

File: lib/helpers/stake/gateway_composer/StakeHelper.js (hunk 2 above)
Comment (hobofan) [reply lib/helpers/stake/gateway_composer/StakeHelper.js]: They are being passed into the constructor of `Contract`, so they are still being used. Changing it more is out of the scope of this PR in my opinion (considering that this class is also going to be deprecated).

**Comment by hobofan** [comment]
I saw that `Utils.sendTransaction` also exists as a copy in this repo now, since #99, so this isn't blocked by a new release of mosaic.js anymore. There is a follow-up ticket to switch to the one exposed by mosaic.js: #108

File: lib/helpers/stake/gateway_composer/StakeHelper.js (hunk 1 above)
Comment (hobofan) [reply lib/helpers/stake/gateway_composer/StakeHelper.js]: Same as the other comment (think this is out of scope of the PR).

**Pull request merged**